### PR TITLE
feat(workflow): Support `owner` issue search filter 

### PR DESCRIPTION
**Pull request by iProgramStuff**
Adding support to the issue search for things like `owner:#workflow`, `owner:me`, and `owner:cfuller@sentry.io`.

This uses the new GroupOwner model as a basis.

### CODE REVIEW

File: src/sentry/search/snuba/backend.py [hunk 1]
```diff
@@ -122,6 +123,30 @@ def inbox_filter(inbox, projects):
...
+    if isinstance(owner, Team):
+        return Q(
+            id__in=GroupOwner.objects.filter(
+                team=owner, project_id__in=project_ids, organization_id=organization_id
+            )
+            .values_list("group_id", flat=True)
+            .distinct()
+        )
+
+    else:
```
Comment (markstory) [inline src/sentry/search/snuba/backend.py:140]: You don't need this else as your if block returns. Should this case be wrapped with `if isinstance(owner, User)`?

Comment (iProgramStuff) [reply src/sentry/search/snuba/backend.py:140]: I don't know what I'd do in the case that the owner isn't a User or a Team. Raise an error? Could I just let django raise an error when tying to assign a non-User to a User?

Removed the else

Comment (markstory) [reply src/sentry/search/snuba/backend.py]: You could raise an InvalidSearchQuery error.

Comment (iProgramStuff) [reply src/sentry/search/snuba/backend.py]: Done and done. Thanks.

File: src/sentry/api/issue_search.py [hunk 2]
```diff
@@ -27,6 +27,7 @@ class IssueSearchVisitor(SearchVisitor):
         "assigned_to": ["assigned"],
         "bookmarked_by": ["bookmarks"],
         "subscribed_by": ["subscribed"],
+        "owner": ["owner"],
```
Comment (markstory) [inline src/sentry/api/issue_search.py:30]: This key won't show up in the autocompletes unless you also update `static/app/stores/tagStore.tsx`. The `getIssueAttributes()` will need to know about this field before you can wire up autocomplete to it. You'll likely also need to special case the value completion to use the team/memberlist.

File: src/sentry/search/snuba/backend.py [hunk 3]
```diff
@@ -122,6 +123,31 @@ def inbox_filter(inbox, projects):
     return query
 
 
+def owner_filter(owner, projects):
+    from sentry.models import Team, User
```
Comment (wedamija) [inline src/sentry/search/snuba/backend.py:127]: Nit: Let's import these at the top.

**Review by wedamija — APPROVED** [review]
lgtm, but one of your tests is busted. Agree we should add this to js so that we get autocomplete.

File: src/sentry/search/snuba/backend.py [hunk 4]
```diff
@@ -122,6 +123,31 @@ def inbox_filter(inbox, projects):
...
+    if isinstance(owner, Team):
+        return Q(
+            id__in=GroupOwner.objects.filter(
+                team=owner, project_id__in=project_ids, organization_id=organization_id
+            )
+            .values_list("group_id", flat=True)
+            .distinct()
+        )
+    elif isinstance(owner, User):
+        return Q(
```
Comment (davidenwang) [inline src/sentry/search/snuba/backend.py:140]: if we want `owner:me`/`owner:<user>` to include team info like the assigned filter, do we also have to do a check for teams that are assigned to the user, or is that already handled here?

File: src/sentry/api/issue_search.py (hunk 2 above)
Comment (iProgramStuff) [reply src/sentry/api/issue_search.py:30]: Thanks Mark - and David for tackling the frontend bits :-)

File: src/sentry/search/snuba/backend.py (hunk 4 above)
Comment (iProgramStuff) [reply src/sentry/search/snuba/backend.py:153]: Yeah, at first I wasn't sure if `owner:dwang@sentry.io` was going to include the team for that user - or if that was specific to `owner:me`. If it was specific, I think I would have needed to change the parser as it sends a User object whether you type `me` or an email and I'd have no way to distinguish. 

We decided to include the teams of a User for all owner searches and I added some logic to this function to handle that.

**Pull request merged**